lib/metrics/metrics.go

```diff
@@ -24,45 +24,113 @@ import (
 	"go.k6.io/k6/stats"
 )
 
-// TODO: refactor this, using non thread-safe global variables seems like a bad idea for various reasons...
-
-//nolint:gochecknoglobals
-var (
-	// Engine-emitted.
-	VUs               = stats.New("vus", stats.Gauge)
-	VUsMax            = stats.New("vus_max", stats.Gauge)
-	Iterations        = stats.New("iterations", stats.Counter)
-	IterationDuration = stats.New("iteration_duration", stats.Trend, stats.Time)
-	DroppedIterations = stats.New("dropped_iterations", stats.Counter)
-	Errors            = stats.New("errors", stats.Counter)
+const (
+	VUsName               = "vus" //nolint:golint
```

Comment:
I think I've mentioned this before, but I don't see a good reason for these to be public, or even consts. They can be simple strings in `RegisterBuiltinMetrics()`, because they shouldn't have to be directly referenced anywhere... They should only be used once, in the definition, and never needed as strings anywhere else again.